includes/init_includes/init_coupons.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Try to match the HTTP_REFERER header against our coupon records.
+ *
+ * @return ?string the coupon_code if found, else null.
+ */
+function initCouponReferrerCheck(): ?string {
+    global $db;
+
+    if (empty($_SERVER['HTTP_REFERER'])) {
+        // There is no referer header, cannot do any lookup.
+        return null;
+    }
+    $referrer = $_SERVER['HTTP_REFERER'];
+
+    // Check for coupon that probably matches this referer.  More expensive validation done later.
```

Comment:
Extensive?

Reply:
I refer to the loading of lang strings and instantiation of ot_coupon as being expensive.  Cheap validation is to directly query `coupons` table without regard to whether the ot_coupon module is enabled.